game.cpp

```diff
@@ -1351,18 +1326,72 @@ void test_game() //!OCLINT tests may be many
     assert(player_copy.get_speed() == p.get_speed());
   }
 
-  /// When a player is killed it stays in the player vector but its state is dead
+  /// When a player is out it stays in the player vector but its state is out
   {
     game g;
 
     auto num_of_players_begin = g.get_v_player().size();
 
-    //kill the first player
     g.kill_player(0);
 
     assert(num_of_players_begin == g.get_v_player().size());
-    assert(is_dead(g.get_player(0)));
+    assert(is_out(g.get_player(0)));
+  }
+
+#ifdef FIX_ISSUE_606
+  {
+    // (626) A player that is out can be revived
+    game g;
+    g.kill_player(0);
+    assert(is_out(g.get_player(0)));
+    g.revive_player(0);
+    assert(!is_out(g.get_player(0)));
+  }
+#endif
+
+#ifdef FIX_ISSUE_611
+  {
+    // (611) A player that is out revives after some time
+    game g;
+    g.kill_player(0);
+    const int revive_time = 100;
+    for (int i = 0; i < revive_time; ++i)
+      {
+        assert(is_out(g.get_player(0)));
+        g.tick();
+      }
+    assert(!is_out(g.get_player(0)));
+  }
+#endif
+
+#ifdef FIX_ISSUE_606
+  {
+    // (606) When a player goes under some size, it is out
+    game g;
+    player p = g.get_player(0);
```

Comment:
I see the idea here and I agree with that.

However, there is a problem here: the `player p` is a **copy** of the first player in the game. This means that `g.tick();` will not operate on `p`. I predict that, due to this, this test does not even fail.

Solutions are to use either `player& p = g.get_player(0);` or use `g.get_player(0)`/`get_first_player(g)` every time.

Reply:
Oops, indeed! I should know that. Conveniently, `get_player` returns a reference to the player 😏